diff -uNdr lives-1.4.7_orig/lives-plugins/plugins/decoders/asf_decoder.c lives-1.4.7/lives-plugins/plugins/decoders/asf_decoder.c
--- lives-1.4.7_orig/lives-plugins/plugins/decoders/asf_decoder.c	2011-09-18 23:31:04.000000000 +0400
+++ lives-1.4.7/lives-plugins/plugins/decoders/asf_decoder.c	2011-10-17 13:35:51.762583340 +0400
@@ -1227,7 +1227,7 @@
 	snprintf(cdata->audio_name,16,"unknown");
 
       } else if (type == LIVES_MEDIA_TYPE_VIDEO) {
-	priv->st->codec->codec_type = CODEC_TYPE_VIDEO;
+	priv->st->codec->codec_type = AVMEDIA_TYPE_VIDEO;
 	if (vidindex!=-1&&vidindex!=priv->st->id) {
 	  fprintf(stderr, "asf_decoder: unhandled multiple vidstreams %d and %d in %s\n",vidindex,priv->st->id,cdata->URI);
 	  got_vidst=TRUE;
